pkg/crypto/signaturehash/signaturehash.go

```diff
@@ -40,8 +40,12 @@ func Algorithms() []Algorithm {
 
 // SelectSignatureScheme returns most preferred and compatible scheme.
 func SelectSignatureScheme(sigs []Algorithm, privateKey crypto.PrivateKey) (Algorithm, error) {
+	signer, ok := privateKey.(crypto.Signer)
```

Comment:
Is it possible to still accept a `crypto.PrivateKey`? The documentation suggests that the stdlib `crypto.PrivateKey` implementations satisfy `crypto.Signer` but that's not necessarily true for custom implementations. I think this would be a breaking API change if someone provides a custom implementation that does not satisfy `crypto.Signer`?

Reply:
Either it should fail because everyone should be implementing `crypto.Signer` underneath - the `crypto.PrivateKey` was strictly there for backwards compatibility purposes.

Reply:
A custom implementation written currently would never work in the first place, because ultimately pion/dtls would invoke Go's standard crypto APIs that assumes `crypto.PrivateKey` satisfies `crypto.Signer`.